numpy/polynomial/tests/test_chebyshev.py

```diff
@@ -214,6 +214,9 @@ def test_chebint(self):
         assert_raises(ValueError, cheb.chebint, [0], .5)
         assert_raises(ValueError, cheb.chebint, [0], -1)
         assert_raises(ValueError, cheb.chebint, [0], 1, [0, 0])
+        assert_raises(TypeError, cheb.chebint, [0], lbnd=[0])
+        assert_raises(TypeError, cheb.chebint, [0], scl=[0])
+        assert_raises(ValueError, cheb.chebint, [0], axis=.5)
```

Comment:
This strikes me as backwards. The first two should be `ValueError`s, and the second should be a `TypeError`.

I don't think there is any precedent in numpy for "wrong number of dimensions" &rarr; `TypeError`, but there absolutely is precedent for `isinstance(axis, float)` &rarr; `TypeError`

Reply:
Floats here are fine as long as they are integer valued, same with cnt. Might not be the way things are done now, but this is very old code. What is done is conversion to `int` followed by a  check for equality against the original value.

If you think the first two should be `ValueError`, then the original was correct. However, the ndim check is for scalar, so passing a list could be considered a `TypeError`.

Reply:
I see your point - the `int(x) == x` check is consistent with `ValueError`, even if neither are consistent with the axis argument in the rest of numpy.

So yes, I prefered the original - sorry for not being clearer there.